File: bit_manipulation/0137_single_number_ii/0137_single_number_ii.py
class Solution(object):
    def singleNumber(self, A):
        one, two, carry = 0, 0, 0
        for x in A:
            two |= one & x
            one ^= x
            carry = one & two
            one &= ~carry
            two &= ~carry
        return one

A = [2,2,3,2]
res = Solution().singleNumber(A)
print(res)

File: bit_manipulation/0137_single_number_ii/0137_single_number_ii.go
package main 

import "fmt"

func singleNumber(nums []int) int {
	ones, twos := 0, 0
	for i:=0;i<len(nums);i++ {
		ones = (ones ^ nums[i]) & ^twos
		twos = (twos ^ nums[i]) & ^ones
	}

	return ones
}

func main() {
	nums := []int{2,2,3,2}
	res := singleNumber(nums)
	fmt.Println(res)
}